### PR TITLE
pythonPackages.bsdiff4: init at 1.2.0

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change
New package. Requires new license "BSD-Protection" (see https://spdx.org/licenses/BSD-Protection.html)

Tests enabled. Includes cli tool.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/bsdiff4/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  checkPhase = ''
+    mv bsdiff4 _bsdiff4
+    python -c 'import bsdiff4; bsdiff4.test()'
+  '';
+
+  meta = with stdenv.lib; {
+    description = "binary diff and patch using the BSDIFF4-format";
+    homepage = https://github.com/ilanschnell/bsdiff4;
```
Comment (jonringer) [inline pkgs/development/python-modules/bsdiff4/default.nix:23]: rfc 45
```suggestion
    homepage = "https://github.com/ilanschnell/bsdiff4";
```

Comment (risicle) [reply pkgs/development/python-modules/bsdiff4/default.nix]: Damnit yes. Fixed now.

**Comment by risicle** [comment]
@GrahamcOfBorg build python37Packages.bsdiff4

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 94622 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary>  - python27Packages.bsdiff4  - python37Packages.bsdiff4  - python38Packages.bsdiff4</details>

**Pull request merged**